src/main/java/com/google/firebase/database/snapshot/NodeUtilities.java

```diff
@@ -58,6 +60,10 @@ public static Node NodeFromJSON(Object value, Node priority) throws DatabaseExce
         return new DoubleNode((Double) value, priority);
       } else if (value instanceof Boolean) {
         return new BooleanNode((Boolean) value, priority);
+      } else if (value instanceof BigDecimal) {
+        return new DoubleNode(((BigDecimal) value).doubleValue(), priority);
```

Comment:
This is a lossy conversion. What happens when the BigDecimal is too large to fit in a double?

Reply:
I think BigDecimals should be serialized to String. The entire point of using BigDecimals is to avoid the rounding errors inherent in floating point representation.